src/pytest_inmanta_lsm/lsm_project.py

```diff
@@ -926,81 +936,99 @@ def add_service(
-            service = self.get_service(srv)
+        :param service: The service to validate.
+        :param model: The model to run the compile with.
+        """
+        # Get the model to use for the compile
+        model = self._get_model(model)
 
-        env: dict[str, str] = {}
-        if service_ids:
-            env[inmanta_lsm.const.ENV_INSTANCE_ID] = " ".join(service_ids)
+        # Get the service's current version
+        service = self.get_service(service_id)
 
-        if validation:
-            if len(service_ids) != 1:
-                raise Exception(
-                    f"when performing a validation compile, only one service id can be passed, got {repr(service_ids)}"
-                )
+        # Collect the environment variables that will need to be set for the compile
+        env = {
+            inmanta_lsm.const.ENV_INSTANCE_VERSION: str(service.version),
+            inmanta_lsm.const.ENV_MODEL_STATE: inmanta_lsm.model.ModelState.candidate.value,
+            inmanta_lsm.const.ENV_INSTANCE_ID: str(service_id),
+            inmanta_lsm.const.ENV_PARTIAL_COMPILE: str(self.partial_compile),
+        }
 
-            # Get the service's current version to set it in the env var
-            service = self.get_service(service_ids[0])
-            env[inmanta_lsm.const.ENV_INSTANCE_VERSION] = str(service.version)
+        LOGGER.debug(
+            "Triggering validating compile for service %s with the following environment variables: %s",
+            service_id,
+            env,
+        )
+        with pytest.MonkeyPatch.context() as m:
+            for k, v in env.items():
+                m.setenv(k, v)
+            self.project.compile(model, no_dedent=False)
 
-            # If we have a validation compile, we need to set an additional env var
-            env[inmanta_lsm.const.ENV_MODEL_STATE] = inmanta_lsm.model.ModelState.candidate
+    def exporting_compile(
+        self,
+        service_ids: collections.abc.Sequence[str | uuid.UUID] | None = None,
+        *,
+        model: str | None = None,
+    ) -> None:
+        """
+        Perform an exporting compile for the given services, or all services present in the inventory
+        if no services is specified.
+        Use the provided model for the compilation.  If none is provided, defaults to the model used
+        to export the service definition if there is any, fails otherwise.
+        """
+        # Get the model to use for the compile
+        model = self._get_model(model)
 
-        try:
-            env[inmanta_lsm.const.ENV_PARTIAL_COMPILE] = str(self.partial_compile and bool(service_ids))
-        except AttributeError:
-            # This attribute only exists for iso5+, iso4 doesn't support partial compile.
-            # We then simply don't set the value.
-            if self.partial_compile:
-                warnings.warn("Partial compile is enabled but it is not supported, it will be ignored.")
+        if service_ids is None:
```

Comment:
You could also do a full compile with the service id set? (it can be considered a bit weird, but that may be the point of the test?)

Reply:
The issue I have with it is that doing this really doesn't mean something (it is an undefined scenario) so if someone requests this, they probably don't know what they are requesting (or if they do, they might not realize that the testing library doesn't consider this to be a use case)